com.cuperino.qprompt.json

```diff
@@ -0,0 +1,40 @@
+{
+   "command": "qprompt",
+   "finish-args": [
+      "--share=ipc",
+      "--socket=x11",
+      "--socket=wayland",
+      "--filesystem=host",
+      "--share=network",
+      "--device=all"
+   ],
+   "id": "com.cuperino.qprompt",
+   "cleanup" : [
+      "/include",
+      "/lib/pkgconfig",
+      "/man",
+      "/share/doc",
+      "/share/gtk-doc",
+      "/share/man",
+      "/share/pkgconfig",
+      "*.la",
+      "*.a"
+   ],
+   "modules": [
+      {
+         "buildsystem": "cmake-ninja",
+         "name": "qprompt",
+         "sources": [
+            {
+               "type": "git",
+               "url": "https://github.com/Cuperino/QPrompt.git",
+               "tag": "v1.0.0",
+               "commit": "c886f9b692dc8900af69f01b6af26f19f220090f"
+            }
+         ]
+      }
+   ],
+   "runtime": "org.kde.Platform",
+   "runtime-version": "5.15",
```

Comment:
```suggestion
   "runtime-version": "5.15-21.08",
```

Reply:
I don't know why, but QPrompt silently fails to start when use the 5.15-21.08 runtime instead of just 5.15...

Reply:
you mean it's crashing on startup? if you run the flatpak with `--command=sh` and `--devel` you can start the executable in gdb and figure out a stack trace.

Reply:
This is strange... When I get a shell inside the application's sandbox `flatpak --command=sh --devel run com.cuperino.qprompt` and run QPrompt I can see that it clearly is crashing with a segmentation fault. But when I get a shell inside an application sandbox without having to install it `flatpak-builder --run ./build com.cuperino.qprompt.json sh`, QPrompt runs almost perfectly and only icons are missing. I hypothesize these might be separate issues and the crash may be related to permissions. Or this may be related to [this CMake install bug which I thought was specific to Windows](https://bugs.kde.org/show_bug.cgi?id=445248), but I doubt it.

I haven't been able to run the debugger with the first approach because the build always lacks debug symbols irregardless of what I try. I've tried setting `"config-opts": "--DCMAKE_BUILD_TYPE=RelWithDebInfo",` inside the module in the manifest, and defining `CMAKE_BUILD_TYPE=Debug` before performing the build.

Could someone point me to the correct way of building the Flatpak with debug symbols?

Reply:
`flatpak-builder --run` as less syscall filtering than `flatpak run`

And `--devel` still grant a few more than without.

`RelWithDebInfo` build with debug symbols, but there is a `.Debug` package that you need to install.